src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -20,18 +35,169 @@ import {CDK_MENU, Menu} from './menu-interface';
+    this._keyManager.setLastItemActive();
+  }
+
+  /**
+   * Handle keyboard events, specifically changing the focused element and/or toggling the active
+   * items menu.
+   * @param event the KeyboardEvent to handle.
+   */
+  _handleKeyEvent(event: KeyboardEvent) {
+    switch (event.keyCode) {
+      case UP_ARROW:
+      case DOWN_ARROW:
+      case LEFT_ARROW:
+      case RIGHT_ARROW:
+        const horizontalArrows = event.keyCode === LEFT_ARROW || event.keyCode === RIGHT_ARROW;
+        // For a horizontal menu if the left/right keys were clicked, or a vertical menu if the
+        // up/down keys were clicked: if the current menu is open, close it then focus and open the
+        // next  menu.
+        if (
+          (this._isHorizontal() && horizontalArrows) ||
+          (!this._isHorizontal() && !horizontalArrows)
+        ) {
+          event.preventDefault();
+
+          const prevIsOpen = this._keyManager.activeItem?.isMenuOpen();
+          this._keyManager.activeItem?.getMenuTrigger()?.closeMenu();
+
+          this._keyManager.setFocusOrigin('keyboard');
+          this._keyManager.onKeydown(event);
+          if (prevIsOpen) {
+            this._keyManager.activeItem?.getMenuTrigger()?.openMenu();
+          }
+        }
+        break;
+
+      case HOME:
+      case END:
+        if (!hasModifierKey(event)) {
+          event.preventDefault();
+          event.keyCode === HOME ? this.focusFirstItem('keyboard') : this.focusLastItem('keyboard');
+        }
+        break;
+
+      case ESCAPE:
+        event.preventDefault();
+        this._keyManager.activeItem?.getMenuTrigger()?.closeMenu();
+        break;
+
+      case TAB:
+        this._keyManager.activeItem?.getMenuTrigger()?.closeMenu();
+        break;
+
+      default:
+        this._keyManager.onKeydown(event);
+    }
+  }
+
+  /** Setup the FocusKeyManager with the correct orientation for the menu bar. */
+  private _setKeyManager() {
+    this._keyManager = new FocusKeyManager(this._allItems).withWrap().withTypeAhead();
+
+    if (this._isHorizontal()) {
+      this._keyManager.withHorizontalOrientation(this._dir.value);
+    } else {
+      this._keyManager.withVerticalOrientation();
+    }
+  }
+
+  /** Subscribe to the MenuStack close and empty observables. */
+  private _subscribeToMenuStack() {
+    this._menuStack.close().subscribe((item: MenuStackItem) => this._closeOpenMenu(item));
+    this._menuStack.empty().subscribe((event: FocusNext) => this._toggleOpenMenu(event));
+  }
+
+  /**
+   * Close the open menu if the current active item opened the requested MenuStackItem.
+   * @param item the MenuStackItem requested to be closed.
+   */
+  private _closeOpenMenu(item: MenuStackItem) {
+    if (item === this._keyManager.activeItem?.getMenu()) {
+      this._keyManager.activeItem.getMenuTrigger()?.closeMenu();
+      this._keyManager.setFocusOrigin('keyboard');
+      this._keyManager.setActiveItem(this._keyManager.activeItem);
```

Comment:
Repeating `this._keyManager` won't minify with anything aside from Closure compiler. If you have it multiple times within the same function, it's better to save it in a variable.